### ISSUE TITLE
Using Sprig with Set Password Form

### ISSUE THREAD

**Issue by kylecotter** [putyourlightson/craft-sprig#300]
I'm attempting to use Sprig on a User Account Set Password Form to prevent a hard page refresh and gracefully display validation errors as needed when a front-end user sets a new password.

* User sign in and password reset request forms work as expected with Sprig.
* The set password form however, upon submission, will timeout or exhaust all PHP memory on the ajax request if there are validation errors in the `newPassword` submitted. The error message is never returned.
* A password that meets validation will save without issue.
* This all works without issue in a non-sprig context.

Simple reproduction case:

In your Craft designated "set password" template:
```
{{ sprig('_sprig/account/set-password', {
    code: code,
    id: id,
}) }}
```

In `_sprig/account/set-password`:

```
<div id="replace">
    {% if errors is defined %}
        {{ errors|join(', ') }}
    {% elseif success is defined and success %}
        {% do sprig.location('/') %}
    {% endif %}
</div>

<form
    sprig
    s-method="post"
    s-action="users/set-password"
    s-trigger="submit"
    s-replace="#replace",
    accept-charset="UTF-8"
>

    {{ hiddenInput('code', code) }}
    {{ hiddenInput('id', id) }}
    {{ input('password', 'newPassword') }}

    <button type="submit">
        Submit
    </button>
</form>
```
Steps: 
1. Copy a password reset URL from Craft CP on a non-admin user
2. Upon typing `123` as the new password and submitting, inspecting the Network request will show a never ending request
3. Upon typing `1234Abcd!` and submitting, you should be successful in completing the request

Versions: 
Craft CMS 4.4.3
Sprig 2.5.1

PHP 8.1.16
MySQL 8.0.32

Am I doing anything obviously wrong here?

Thanks!

**Comment by bencroker** [comment]
This is a tricky one, due to the way Craft handles the set password action on failure (nothing that you’re doing wrong). Request handling will change in v3 as per https://github.com/putyourlightson/craft-sprig/issues/301 but until then I’m going to have to add special case handling for this. I’ll be able to work on that later this week and will let you know when it’s ready.

**Comment by bencroker** [comment]
Fixed in https://github.com/putyourlightson/craft-sprig-core/commit/5ed215bf493a94f48c94ea226271fcae14a0ff12 and released in Sprig Core 2.4.2 which should now “just work” after doing a `composer update`.
